### PR TITLE
Metadata and encoding reading

### PR DESCRIPTION
**Pull request by jsaarimaa**
## Contents
1. Implementation for reading the matadata from a stream.
2. Implementation for parsing the encoding from the a stream
3. Unit tests for the implemented public functionality

### CODE REVIEW

File: Px.Utils.UnitTests/PxFileTests/PxFileMetadataReaderTests/GetEncodingAsyncTests.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+﻿using Px.Utils.UnitTests.PxFileTests.Fixtures;
+using PxUtils.PxFile.Meta;
```
Comment (sakari-malkki) [inline Px.Utils.UnitTests/PxFileTests/PxFileMetadataReaderTests/GetEncodingAsyncTests.cs:2]: This namespace has a different naming convention compared to the one above it. PxUtils and Px.Utils.

File: Px.Utils/PxFile/Exceptions/InvalidPxFileMetadataException.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Px.Utils.PxFile.Exceptions
```
Comment (sakari-malkki) [inline Px.Utils/PxFile/Exceptions/InvalidPxFileMetadataException.cs:1]: Here Px.Utils

File: Px.Utils.UnitTests/PxFileTests/Fixtures/MinimalPx.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿namespace Px.Utils.UnitTests.PxFileTests.Fixtures
```
Comment (sakari-malkki) [inline Px.Utils.UnitTests/PxFileTests/Fixtures/MinimalPx.cs:1]: Px.Utils here

**Comment by sakari-malkki** [comment]
Px.Utils.UnitTests.PxFileTests.Fixtures and Px.Utils.PxFile.Exceptions namespaces are named "Px.Utils" unlike the rest of them.

**Review by sakari-malkki — APPROVED** [review]
Left comments to be resolved about namespace naming inconsistencies. Apart from that looks good to go.

File: Px.Utils/PxFile/Exceptions/InvalidPxFileMetadataException.cs (hunk 2 above)
Comment (jsaarimaa) [reply Px.Utils/PxFile/Exceptions/InvalidPxFileMetadataException.cs]: Fixed

File: Px.Utils.UnitTests/PxFileTests/Fixtures/MinimalPx.cs (hunk 3 above)
Comment (jsaarimaa) [reply Px.Utils.UnitTests/PxFileTests/Fixtures/MinimalPx.cs]: Fixed

**Review by sakari-malkki — APPROVED** [review]

**Pull request merged**